loom/engine/bindings/loom/lmSQLite.cpp

```diff
@@ -0,0 +1,384 @@
+    }
+      
+    int getlastInsertRowId()
+    {
+        //"NOTE: In SQLite the row ID is a 64-bit integer but for all practical 
+        //database sizes you can cast the 64 bit value to a 32-bit integer."
+        //
+        // - Some Guy on The Internet
+        //
+        int rowid = (int)sqlite3_last_insert_rowid(dbHandle);
+        return rowid;
+    }
+
+    void close()
+    {
+        //close the database
+        int res = sqlite3_close_v2(dbHandle);
+        if(res != SQLITE_OK)
+        {
+            lmLogError(gSQLiteGroup, "Error closing the SQLite database: %s with message: %s", szDBName, getErrorMessage());
+        }
+    }
+};
+
+
+
+//SQLite Statement binding for Loomscript
+class Statement
+{
+private:
+    Connection *parentDB;
+    static char szReturnString[4096];
+public:
+    sqlite3_stmt *statementHandle;
+
+
+    Statement(Connection *c)
+    {
+        parentDB = c;
+    }
+
+    int getParameterCount()
+    {
+        return sqlite3_bind_parameter_count(statementHandle);
+    }
+
+    const char *getParameterName(int index)
+    {
+        const char *name = sqlite3_bind_parameter_name(statementHandle, index);
+        if(name == NULL)
+        {
+            lmLogError(gSQLiteGroup, "Invalid index for getParameterName in database: %s", parentDB->getDBName());
+            return NULL;
+        }
+
+        //copy the string into our static return array so it survives the trip to Loomscript Land!
+        strcpy(szReturnString, name);
+        return szReturnString;
+    }
+
+    int getParameterIndex(const char* name)
+    {
+        int index = sqlite3_bind_parameter_index(statementHandle, name);
+        if(name == NULL)
+        {
+            lmLogError(gSQLiteGroup, "Invalid name for getParameterIndex in database: %s", parentDB->getDBName());
+        }
+        return index;
+    }
+
+    int bindInt(int index, int value)
+    {
+        int result = sqlite3_bind_int(statementHandle, index, value);
+        if(result != SQLITE_OK)
+        {
+            lmLogError(gSQLiteGroup, "Error calling bindInt for database: %s with Result Code: %i", parentDB->getDBName(), result);
+        }
+        return result;        
+    }
+
+    int bindDouble(int index, double value)
+    {
+        int result = sqlite3_bind_double(statementHandle, index, value);
+        if(result != SQLITE_OK)
+        {
+            lmLogError(gSQLiteGroup, "Error calling bindDouble for database: %s with Result Code: %i", parentDB->getDBName(), result);
+        }
+        return result;        
+    }
+
+    int bindString(int index, const char *value)
+    {
+        int result = sqlite3_bind_text(statementHandle, index, value, -1, SQLITE_TRANSIENT); 
+        if(result != SQLITE_OK)
+        {
+            lmLogError(gSQLiteGroup, "Error calling bindString for database: %s with Result Code: %i", parentDB->getDBName(), result);
+        }
+        return result;        
+    }
+
+
+    int bindBytes(lua_State *L)
```

Comment:
I just copied the other 2 uses of ByteArray between LS and native and this
is how they.did it. No other code does a straight 1:1.
On 12 Feb 2015 5:33 AM, "Ben Garney" <notifications@github.com> wrote:

> In loom/engine/bindings/loom/lmSQLite.cpp
> <https://github.com/LoomSDK/LoomSDK/pull/261#discussion_r24556196>:
>
> > +        }
> > +        return result;
> > +    }
> > +
> > +    int bindString(int index, const char *value)
> > +    {
> > +        int result = sqlite3_bind_text(statementHandle, index, value, -1, SQLITE_TRANSIENT);
> > +        if(result != SQLITE_OK)
> > +        {
> > +            lmLogError(gSQLiteGroup, "Error calling bindString for database: %s with Result Code: %i", parentDB->getDBName(), result);
> > +        }
> > +        return result;
> > +    }
> > +
> > +
> > +    int bindBytes(lua_State *L)
>
> Can't pass utByteArray?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/LoomSDK/LoomSDK/pull/261/files#r24556196>.
>